pf4j/src/main/java/org/pf4j/ServiceProviderExtensionFinder.java

```diff
@@ -94,18 +95,25 @@ public ServiceProviderExtensionFinder(PluginManager pluginManager) {
             final Set<String> bucket = new HashSet<>();
 
             try {
-                URL url = ((PluginClassLoader) plugin.getPluginClassLoader()).findResource(getExtensionsResource());
-                if (url != null) {
-                    Path extensionPath;
-                    if (url.toURI().getScheme().equals("jar")) {
-                        extensionPath = FileUtils.getPath(url.toURI(), getExtensionsResource());
-                    } else {
-                        extensionPath = Paths.get(url.toURI());
-                    }
-
-                    bucket.addAll(readExtensions(extensionPath));
-                } else {
+                Enumeration<URL> urls = ((PluginClassLoader) plugin.getPluginClassLoader()).findResources(getExtensionsResource());
+                if (urls == null || !urls.hasMoreElements()) {
                     log.debug("Cannot find '{}'", getExtensionsResource());
+                } else {
+                    while (urls.hasMoreElements()) {
+                        URL url = urls.nextElement();
+                        if (url != null) {
```

Comment:
The same thing (no need for null check)

Reply:
You're right. This null check is definitely not necessary. I've [updated the pull request](https://github.com/pf4j/pf4j/pull/262/commits/23a4263eef7de59ed4a5bb29f84af4010b749d75) accordingly.